istioctl/cmd/istioctl/main.go

```diff
@@ -143,26 +143,17 @@ var (
 		Short:             "Istio control interface",
 		SilenceUsage:      true,
 		DisableAutoGenTag: true,
-		Long: `
-Istio configuration command line utility.
-
-Create, list, modify, and delete configuration resources in the Istio
-system.
-
-Available routing and traffic management configuration types:
-
-	[virtualservice gateway destinationrule serviceentry httpapispec httpapispecbinding quotaspec quotaspecbinding servicerole servicerolebinding policy]
-
-See https://istio.io/docs/reference/ for an overview of Istio routing.
-
+		Long: `Istio configuration command line utility for service operators to
+debug and diagnose their Istio mesh.
 `,
 		PersistentPreRunE: istioPersistentPreRunE,
 	}
 
 	postCmd = &cobra.Command{
-		Use:     "create",
-		Short:   "Create policies and rules",
-		Example: "istioctl create -f example-routing.yaml",
+		Use:        "create",
+		Deprecated: "Use `kubectl create` instead (see https://kubernetes.io/docs/tasks/tools/install-kubectl)",
+		Short:      "Create policies and rules",
+		Example:    "istioctl create -f example-routing.yaml",
 		RunE: func(c *cobra.Command, args []string) error {
```

Comment:
We also lose validation for cases where there is no kubectl. Also not sure how this will affect the test stuff we are doing with a local pilot. Will this end up mandating a k8s for everyone?

Reply:
Not sure I follow. What are the cases where we couldn't use kubectl instead of istioctl? The local pilot tests read directly from files. istioctl isn't involved in that flow at all today.

AFAICT, CF is integrating directly with Pilot (via copilot), don't use istioctl, and has their own validation logic. We've discussed reusing validation schemas (e.g. via proto annotations) across platforms, but that is unrelated to kubectl vs. istioctl. 

We're using kube-apiserver in all other cases and config validation is performed server-side via a ValidatingWebhook.

Reply:
Consul and on-prem is not using kube-apiserver

Reply:
( I mean - the configs where pilot is using etcd and standalone apiserver )